File: core/src/main/java/com/tvsc/core/model/BannerInfo.java
package com.tvsc.core.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.squareup.moshi.Json;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * @author Taras Zubrei
 */
@Data
@NoArgsConstructor
@Accessors(chain = true)
public class BannerInfo implements Comparable<BannerInfo> {
    @JsonProperty("keyType")
    @Json(name = "keyType")
    private String type;
    @JsonProperty("subKey")
    @Json(name = "subKey")
    private String key;
    private String fileName;
    private RatingsInfo ratingsInfo;

    @Override
    public int compareTo(BannerInfo o) {
        return this.ratingsInfo.average.compareTo(o.ratingsInfo.average);
    }

    @Data
    @NoArgsConstructor
    @Accessors(chain = true)
    public static class RatingsInfo {
        private Double average;
        private Long count;
    }
}

File: core/src/main/kotlin/com/tvsc/core/model/BannerInfo.kt
package com.tvsc.core.model

import com.fasterxml.jackson.annotation.JsonProperty

/**
 *
 * @author Taras Zubrei
 */
data class BannerInfo(
        @JsonProperty("keyType") var type: String,
        @JsonProperty("subKey") var key: String? = null,
        @JsonProperty var fileName: String,
        @JsonProperty var ratingsInfo: RatingsInfo
) : Comparable<BannerInfo> {
    override fun compareTo(other: BannerInfo): Int {
        return this.ratingsInfo.average.compareTo(other.ratingsInfo.average);
    }

    data class RatingsInfo(
            @JsonProperty var average: Double,
            @JsonProperty var count: Long
    )
};
